### PR TITLE
Maintenance & build optimization

### PR DESCRIPTION
**Pull request by felixmosh**
- [x] Bump deps to latest
- [x] Clean up deps
- [x] Add css minification on prod
- [x] Enable vendor chunk
- [x] Improves webpack's tree-shaking

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -61,13 +65,31 @@ module.exports = {
...
           },
         ],
       },
     ],
   },
+  optimization: {
+    minimizer: [isProd && `...`, isProd && new CssMinimizerPlugin()].filter(
+      Boolean,
+    ),
+    chunkIds: 'named',
```
Comment (felixmosh) [inline webpack.config.js:80]: Vendor chunk

File: webpack.config.js [hunk 2]
```diff
@@ -61,13 +65,31 @@ module.exports = {
...
+                module: 'esnext',
               },
             },
           },
         ],
       },
     ],
   },
+  optimization: {
+    minimizer: [isProd && `...`, isProd && new CssMinimizerPlugin()].filter(
```
Comment (felixmosh) [inline webpack.config.js:77]: Minify css on prod

File: webpack.config.js [hunk 3]
```diff
@@ -61,13 +65,31 @@ module.exports = {
               transpileOnly: true,
               compilerOptions: {
                 sourceMap: !isProd,
+                module: 'esnext',
```
Comment (felixmosh) [inline webpack.config.js:68]: Improves webpack's tree-shaking

**Review by vcapretz — APPROVED** [review]
nice one!! please make sure you rebase before merging @felixmosh :D

**Pull request merged**